gems/pending/disk/modules/miq_disk_cache.rb

```diff
@@ -0,0 +1,111 @@
+require 'rufus/lru'
+require_relative "../MiqDisk"
+require 'ostruct'
+
+module MiqDiskCache
+  MIN_SECTORS_PER_ENTRY = 32
+  DEF_LRU_HASH_ENTRIES  = 100
+  DEBUG_CACHE_STATS     = false
+
+  attr_reader :d_size, :blockSize, :lru_hash_entries, :min_sectors_per_entry, :cache_hits, :cache_misses
+
+  def self.new(down_stream, lru_hash_entries = DEF_LRU_HASH_ENTRIES, min_sectors_per_entry = MIN_SECTORS_PER_ENTRY)
+    raise "MiqDiskCache: Downstream Disk Module is nil" if down_stream.nil?
+    @dInfo                       = OpenStruct.new
+    @dInfo.lru_hash_entries      = lru_hash_entries
+    @dInfo.min_sectors_per_entry = min_sectors_per_entry
+    @dInfo.block_size            = down_stream.blockSize
+    @dInfo.down_stream           = down_stream
```

Comment:
Hmmm, actually, shouldn't this be `up_stream`? Data flows from upstream to downstream.